app/models/grda_warehouse/hud/client.rb

```diff
@@ -984,16 +984,20 @@ def scope_for_ongoing_residential_enrollments
         ongoing
     end
 
-    def scope_for_other_enrollments
-      service_history_enrollments.
-        entry.
-        hud_non_residential
+    def scope_for_other_enrollments(user)
+      active_consent_model.scope_for_other_enrollments(user)
     end
 
-    def scope_for_residential_enrollments
-      service_history_enrollments.
-        entry.
-        hud_residential
+    def scope_for_residential_enrollments(user)
+      active_consent_model.scope_for_residential_enrollments(user)
+    end
+
+    def active_consent_model
+      @active_consent_model ||= GrdaWarehouse::Config.active_consent_class.new(client: self)
+    end
+
+    def self.active_consent_model
+      @active_consent_model ||= GrdaWarehouse::Config.active_consent_class.new(client: self)
```

Comment:
This won't work (maybe isn't in use?) since `client:` should be an instance of a client, AND it's memoizing at the class level, which is almost never what you want.